nifi-commons/nifi-record/src/test/java/org/apache/nifi/serialization/record/ResultSetRecordSetTest.java

```diff
@@ -157,10 +142,31 @@ public void testCreateSchemaWhenOtherType() throws SQLException {
         assertEquals(RecordFieldType.DECIMAL.getDecimalDataType(30, 10), resultSchema.getField(0).getDataType());
     }
 
+    @Test
+    public void testCreateSchemaWhenOtherTypeAndNoLogicalTypes() throws SQLException {
+        // given
+        final List<RecordField> fields = new ArrayList<>();
+        fields.add(new RecordField("decimal", RecordFieldType.DECIMAL.getDecimalDataType(30, 10)));
+        fields.add(new RecordField("date", RecordFieldType.DATE.getDataType()));
+        fields.add(new RecordField("time", RecordFieldType.TIME.getDataType()));
+        fields.add(new RecordField("timestamp", RecordFieldType.TIMESTAMP.getDataType()));
+        final RecordSchema recordSchema = new SimpleRecordSchema(fields);
+        final ResultSet resultSet = givenResultSetForOther(fields);
+
+        // when
+        final ResultSetRecordSet testSubject = new ResultSetRecordSet(resultSet, recordSchema, 10, 0, false);
+        final RecordSchema resultSchema = testSubject.getSchema();
+
+        // then
+        assertEquals(RecordFieldType.STRING.getDataType(), resultSchema.getField(0).getDataType());
```

Comment:
This checks the type of the first field, but what about the other fields? Should all fields be of type `STRING`?  This should also be checked.

Reply:
That's correct. Thank you for pointing that out! I forgot to add the loop for checking all the fields.